packages/core/src/test/awsService/ec2/commands.test.ts

```diff
@@ -0,0 +1,34 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import assert from 'assert'
+import { Ec2ConnectionManagerMap } from '../../../awsService/ec2/activation'
+import { getConnectionManager } from '../../../awsService/ec2/commands'
+import { Ec2ConnectionManager } from '../../../awsService/ec2/model'
+import { Ec2Selection } from '../../../awsService/ec2/prompter'
+
+describe('getConnectionManager', async function () {
+    let connectionManagers: Ec2ConnectionManagerMap
+
+    beforeEach(function () {
+        connectionManagers = new Map<string, Ec2ConnectionManager>()
+    })
+
+    it('only creates new connection managers once for each region ', async function () {
```

Comment:
this makes me wonder if there is be 1 manager for everything. having many managers seems like an unnecessary problem, and partly defeats the purpose of a "manager", which is to have one place where visibility/handling can happen.

Reply:
I agree, I think the word manager is being overused here. `Manager` also isn't very informative about what each component is responsible for.  I am leaning towards renaming them to directly describe what they "do". 
Ex. `Ec2ConnectionManager` --> `Ec2Connecter`, `Ec2RemoteSessionManager` --> `Ec2SessionTracker`. Does that seem like the right track?